python/FACTORS.md

```diff
@@ -0,0 +1,78 @@
+# GTSAM Python-based factors
+
+One now can build factors purely in Python using the `CustomFactor` factor.
+
+## Theory
```

Comment:
I'd move this to the end and call it "implementation", and add some more meat copied from your excellent comments in the source files

Reply:
done